From d8b201c8498b7d2614e8f282779b72159e25fb72 Mon Sep 17 00:00:00 2001
From: Sumit Singh <sumsingh@nvidia.com>
Date: Fri, 18 Apr 2014 20:19:46 +0530
Subject: [PATCH 20/27] arm64: asm: relaxed.h: defined relaxed.h

Defined a new header file relaxed.h, which contains
arm64 specific macros which will be used to improve
power efficiency of arm64.

bug 1440421

Change-Id: Iee14115490cb16001d5eac9e309ee6e088b88f44
Signed-off-by: Sumit Singh <sumsingh@nvidia.com>
Reviewed-on: http://git-master/r/398323
(cherry picked from commit beb69cd6d2893c36712d4f927e41da0de729d651)
Reviewed-on: http://git-master/r/422212
Reviewed-by: Automatic_Commit_Validation_User
GVS: Gerrit_Virtual_Submit
Reviewed-by: Bharat Nihalani <bnihalani@nvidia.com>
Signed-off-by: Jesse Chan <jc@linux.com>
Signed-off-by: Michele Beccalossi <beccalossi.michele@gmail.com>
---
 arch/arm64/include/asm/relaxed.h | 33 +++++++++++++++++++++++++++++++++
 1 file changed, 33 insertions(+)
 create mode 100644 arch/arm64/include/asm/relaxed.h

diff --git a/arch/arm64/include/asm/relaxed.h b/arch/arm64/include/asm/relaxed.h
new file mode 100644
index 00000000000..08a217934c4
--- /dev/null
+++ b/arch/arm64/include/asm/relaxed.h
@@ -0,0 +1,33 @@
+/*
+ * arm64/include/asm/relaxed.h
+ *
+ * Copyright (c) 2014 NVIDIA Corporation. All rights reserved.
+ * This software is licensed under the terms of the GNU General Public
+ * License version 2, as published by the Free Software Foundation, and
+ * may be copied, distributed, and modified under those terms.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#ifndef _ASM_RELAXED_H_
+#define _ASM_RELAXED_H_
+
+#include <asm/atomic.h>
+
+#ifndef cpu_relaxed_read_short
+#define cpu_relaxed_read_short(p)	ldax16(p)
+#endif
+
+#ifndef cpu_relaxed_read
+#define cpu_relaxed_read(p)		ldax32(p)
+#endif
+
+#ifndef cpu_relaxed_read_long
+#define cpu_relaxed_read_long(p)	ldax64((u64 *)p)
+#endif
+
+#include <asm-generic/relaxed.h>
+#endif /*_ASM_RELAXED_H_*/
-- 
2.11.0

